cms/djangoapps/modulestore_migrator/api/read_api.py

```diff
@@ -0,0 +1,263 @@
+"""
+API for reading information about previous migrations
+"""
+from __future__ import annotations
+
+import typing as t
+from uuid import UUID
+
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import (
+    CourseLocator, LibraryLocator,
+    LibraryLocatorV2, LibraryUsageLocatorV2, LibraryContainerLocator
+)
+from openedx_learning.api.authoring import get_draft_version
+from openedx_learning.api.authoring_models import (
+    PublishableEntityVersion, PublishableEntity, DraftChangeLogRecord
+)
+
+from openedx.core.djangoapps.content_libraries.api import (
+    library_component_usage_key, library_container_locator
+)
+
+from ..data import (
+    SourceContextKey, ModulestoreMigration, ModulestoreBlockMigrationResult,
+    ModulestoreBlockMigrationSuccess, ModulestoreBlockMigrationFailure
+)
+from .. import models
+
+
+__all__ = (
+    'forward_course',
+    'forward_legacy_library',
+    'forward_context',
+    'forward_blocks',
+    'get_migrations',
+    'get_migration_blocks',
```

Comment:
These API methods replace the existing read API, which was:
```
    "is_successfully_migrated",
    "get_migration_info",
    "get_all_migrations_info",
    "get_target_block_usage_keys",
```

The issue with those methods was that:

1. They returned models or untyped dicts rather than type data objects.
2. They did not distinguish between a migration-with-forwarding (i.e., an "authoritative migration") and other migrations (i.e., "non-authoritative migrations", or "imports").

In this new API, the `foward_context` and `forward_blocks` will only return successful authoritative migrations, where as `get_migrations` and `get_migration_blocks` will return all migrations (with some optional filters).

`is_successfully_migrated(blah)` is now written `forward_context(blah) is not None`.